src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -173,6 +166,15 @@ internal IRuntimeClient RuntimeClient
             RegisterSerializationProviders(serializatonProviderOptionsValue.SerializationProviders);
         }
 
+        public void SetApplicationPartManager(ApplicationPartManager applicationPartManager)
```

Comment:
How does this break the dependency cycle? Is it possible to break it in another way so that SerializationManager doesn't need an initialization step?

Reply:
To be honest, logically it does make sense that there is an explicit point in time that we populate the serializer feature from the parts and feature providers, instead of relying on luck in when the container happens to resolve the SM

Reply:
It's deterministic: you cannot access the serialization manager until it's created and the registrations are populated on creation. This change makes it less certain. It's not like SM is populating some state which might already be in use by some other class

Reply:
The issue is that, if a serializer ask for a `SerializationManager`in its constructor, and that we instantiate these serializer in the `SerializationManager`.... boom.

Reply:
@benjaminpetit oh right :( Alternatively we could call the method `Initialize` and store the `ApplicationPartManager` in the ctor so that it's more obvious that it needs initialization before use.

Reply:
My preference is for calling this method `RegisterSerializers` instead and pass the `ApplicationPartManager` in at this point in time, and is never stored by the SM.

In theory in the future we could call this multiple times (not that we will, nor that we should support it now, but would be applicable logically)

Reply:
My intention was that registration occurs exactly once, before usage. As a result, not all of the data structures are protected by locks / concurrent collections.